frontend/packages/ceph-storage-plugin/integration-tests/tests/2-tests/test-expand.scenario.ts

```diff
@@ -0,0 +1,76 @@
+import { clickKebabAction } from '@console/internal-integration-tests/views/crud.view';
+import { click } from '@console/shared/src/test-utils/utils';
+import { testName } from '@console/internal-integration-tests/protractor.conf';
+import { testPVC, testDeployment } from '../../mocks/expand-test-mocks';
+import { requestSizeValue } from '../../views/add-capacity.view';
+import {
+  goToPersistentVolumeClaims,
+  expandButton,
+  createNewPersistentVolumeClaim,
+  expandSizeOption,
+  capacityDropdown,
+} from '../../views/pvc.view';
+import { SIZE_UNITS } from '../../utils/consts';
+
+const expandValue = String(Number(testPVC.size) + 1);
+
+const createDeployment = () =>
+  execSync(`echo '${JSON.stringify(testDeployment)}' | kubectl create -f -`);
+
+describe('Tests Expand flow for PVC', () => {
+  beforeAll(async () => {
+    // Create a PVC
+    await createNewPersistentVolumeClaim(testPVC, true, createDeployment);
+    // Wait for it to bind
+    await goToPersistentVolumeClaims();
+  });
+
+  it('Test PVC can be expanded (In MiBs)', async () => {
+    await clickKebabAction(testPVC.name, 'Expand PVC');
+    await browser.wait(until.visibilityOf(requestSizeValue));
+    await requestSizeValue.sendKeys(expandValue);
+    await click(capacityDropdown);
+    await click(expandSizeOption(SIZE_UNITS.MI));
+    await click(expandButton);
+
+    const pvcJSON = JSON.parse(
+      execSync(`kubectl get pvc ${testPVC.name} -n ${testName} -o json`).toString(),
+    );
+    const requestedStorage = pvcJSON.spec.resources.requests.storage;
+    expect(requestedStorage.trim()).toEqual(`${expandValue}${SIZE_UNITS.MI}`);
+  });
+
+  it('Test PVC can be expanded (In GiBs)', async () => {
+    await goToPersistentVolumeClaims();
+    await clickKebabAction(testPVC.name, 'Expand PVC');
+    await browser.wait(until.visibilityOf(requestSizeValue));
+    await requestSizeValue.sendKeys(expandValue);
+    await click(capacityDropdown);
+    await click(expandSizeOption(SIZE_UNITS.GI));
+    await click(expandButton);
+
+    const pvcJSON = JSON.parse(
+      execSync(`kubectl get pvc ${testPVC.name} -n ${testName} -o json`).toString(),
+    );
+    const requestedStorage = pvcJSON.spec.resources.requests.storage;
+    expect(requestedStorage.trim()).toEqual(`${expandValue}${SIZE_UNITS.GI}`);
+  });
+
+  it('Test PVC can be expanded (In TiBs)', async () => {
+    await goToPersistentVolumeClaims();
+    await clickKebabAction(testPVC.name, 'Expand PVC');
+    await browser.wait(until.visibilityOf(requestSizeValue));
+    await requestSizeValue.sendKeys('1');
+    await click(capacityDropdown);
+    await click(expandSizeOption(SIZE_UNITS.TI));
+    await click(expandButton);
+
+    const pvcJSON = JSON.parse(
+      execSync(`kubectl get pvc ${testPVC.name} -n ${testName} -o json`).toString(),
+    );
+    const requestedStorage = pvcJSON.spec.resources.requests.storage;
+    expect(requestedStorage.trim()).toEqual(`1${SIZE_UNITS.TI}`);
```

Comment:
If `requestedStorage` evaluates to `undefined`, then this will break this line as `trim` will not be available.

Reply:
So? If requested storage is not present then this test should indeed fail.

Reply:
That should not be a problem since that would cause this test to fail. And I am happy with tests failing when `requestedStorage` is not defined.

Reply:
Even checking for `requestedStorage` and if its undefined, the tests will fail and it will give you that the expectation doesn't match instead of giving error `Uncaught TypeError: Cannot read property 'trim' of undefined`. I still prefer checking for undefined as it gives a more clearer message and can help in further debugging.

Reply:
How can `requestedStorage` ever be `undefined`?
When PVC is first created the `requestedStorage`? Passing an undefined value from the frontend will raise an error from the backend. This looks like us being overtly cautious. If for some reason you can set value of requestedStorage < (current Value of requestedStorage) it's a bigger issue. I cannot think of a situation or any possible flow that will make it undefined. Can you think of any?

Reply:
As in this flow `size` is a mandatory parameter from UI, this value will not be undefined.